src/main/java/games/strategy/triplea/settings/SelectionComponentFactory.java

```diff
@@ -169,73 +171,62 @@ public void reset() {
    */
   static Supplier<SelectionComponent<JComponent>> intValueRange(final ClientSetting clientSetting, final int lo,
       final int hi) {
+    return intValueRange(clientSetting, lo, hi, false);
+  }
+
+  /**
+   * Text field that only accepts numbers between a certain range.
+   */
+  static Supplier<SelectionComponent<JComponent>> intValueRange(final ClientSetting clientSetting, final int lo,
+      final int hi, final boolean allowUnset) {
     return () -> new SelectionComponent<JComponent>() {
-      String value = clientSetting.value();
-      final JTextField component = new JTextField(value, String.valueOf(hi).length());
+      private final JSpinner component = new JSpinner(new SpinnerNumberModel(
```

Comment:
The use of a `JSpinner` appears to format the value according to the locale.  For en-US, the port appears with a thousands separator:

![testing-settings](https://user-images.githubusercontent.com/4826349/34451063-a839816a-ece8-11e7-9cac-8ef29191b74b.png)

It's a minor issue, but it was unexpected.  Should we customize the format so no separators are displayed?  Not sure if this is the best solution, but it works:

```diff
diff --git a/src/main/java/games/strategy/triplea/settings/SelectionComponentFactory.java b/src/main/java/games/strategy/triplea/settings/SelectionComponentFactory.java
index 9a2011bd6..0dc76f72d 100644
--- a/src/main/java/games/strategy/triplea/settings/SelectionComponentFactory.java
+++ b/src/main/java/games/strategy/triplea/settings/SelectionComponentFactory.java
@@ -180,8 +180,14 @@ final class SelectionComponentFactory {
   static Supplier<SelectionComponent<JComponent>> intValueRange(final ClientSetting clientSetting, final int lo,
       final int hi, final boolean allowUnset) {
     return () -> new SelectionComponent<JComponent>() {
-      private final JSpinner component = new JSpinner(new SpinnerNumberModel(
-          toValidIntValue(clientSetting.value()), lo - (allowUnset ? 1 : 0), hi, 1));
+      private final JSpinner component = newComponent();
+
+      private JSpinner newComponent() {
+        final JSpinner component = new JSpinner(new SpinnerNumberModel(
+            toValidIntValue(clientSetting.value()), lo - (allowUnset ? 1 : 0), hi, 1));
+        component.setEditor(new JSpinner.NumberEditor(component, "0"));
+        return component;
+      }
 
       @Override
       public JComponent getUiComponent() {
```

Reply:
I don't think there's a need to customize this behaviour. I like it, it makes the number more readable.
And it's purely visual, so you can still enter numbers with or without this seperator and they are properly recognized.

Fun Fact: I Germany Numbers are commonly formatted with commas instead of the decimal point and spaces instead of the "thousands separator" (although the latter one is a question of preference, so you'll find different notations for this, java for example uses dots for the spinner for me ^^).
I found an interesting [article about this](https://en.wikipedia.org/wiki/Decimal_mark#Hindu%E2%80%93Arabic_numeral_system)